test/ccall.jl

```diff
@@ -123,7 +126,11 @@ function test_struct1{Struct}(::Type{Struct})
     b = Float32(123.456)
 
     a2 = copy(a)
-    x = ccall((:test_1, libccalltest), Struct, (Struct, Float32), a2, b)
+    if Struct === Struct1
```

Comment:
It sounds ok for now as long as it doesn't break FunctionWrappers.jl **BEFORE** the alternative is implemented.

In the long term, I think we shouldn't have such restriction as long as it's possible to do efficiently **with** codegen and possible to implement without codegen (with libffi for example). As long as it is expressible, whether or not the construct requires compiler or not shouldn't be visible functionally (i.e. it can of course affect performance). Changing the ccall API to not allow this syntactically is of course different.